crates/cxx-qt-lib/src/core/qtlogging.cpp

```diff
@@ -0,0 +1,7 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qtlogging.h"
```

Comment:
we need the asserts to be put here for checking the size, checking if copy / destructors are trivial (and if they aren't then we need to impl Clone / Drop). And then implementation from the header should be placed here.

Reply:
Done, I think I added the correct check? I need to dig through some of the other trivial types, but I'm not sure what's exactly needed